### PR TITLE
fix: handle the case that base path not started by '/'

### PR DESCRIPTION
**Pull request by fenxu**
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5562)

### CODE REVIEW

**Comment by live1206** [comment]
we need to add a spec test for this case.
Is there a real usage repo for reference?

File: src/docfx/config/ops/OpsConfigAdapter.cs [hunk 1]
```diff
@@ -153,8 +153,7 @@ private async Task<string[]> GetXrefMaps(string xrefMapApiEndpoint, string tag,
         {
             var url = $"{xrefMapApiEndpoint}/v1/xrefmap{tag}{xrefMapQueryParams}";
             var response = await Fetch(url, value404: "{}");
-            return JsonConvert.DeserializeAnonymousType(response, new { links = new[] { "" } }).links
-                ?? Array.Empty<string>();
+            return JsonConvert.DeserializeAnonymousType(response, new { links = new[] { "" } }).links;
```
Comment (live1206) [inline src/docfx/config/ops/OpsConfigAdapter.cs]: revert this line, there was a bug fix for this

File: src/docfx/config/ops/OpsConfigAdapter.cs [hunk 2]
```diff
@@ -105,7 +105,7 @@ private async Task<string> GetBuildConfig(Uri url)
             var xrefMapApiEndpoint = GetXrefMapApiEndpoint(xrefEndpoint);
             if (docset.base_path != "/")
             {
-                xrefQueryTags.Add(docset.base_path);
+                xrefQueryTags.Add($"/{docset.base_path.TrimStart('/')}");
```
Comment (live1206) [inline src/docfx/config/ops/OpsConfigAdapter.cs]: we could not have a spec test for this since this is based on OP api docset info, it should be fine then

Comment (yufeih) [inline src/docfx/config/ops/OpsConfigAdapter.cs]: There is a dedicated type `BasePath` for this purpose

**Pull request merged**